compiler/src/dotty/tools/dotc/core/ConstraintHandling.scala

```diff
@@ -864,6 +864,8 @@ trait ConstraintHandling {
       try bound match
         case bound: TypeParamRef if constraint contains bound =>
           addParamBound(bound)
+        case bound: TypeVar if constraint.contains(bound) =>
```

Comment:
The comment above `addConstraint` says that `bound` should not be a TypeVar, so it looks like a precondition is being violated if this is needed.

Reply:
Ah, thank you.  I force pushed a subset of the commit, to check it didn't independently break anything.  The TypeVar I'm finding is nested in a bound AndType, not top level.